src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -54,11 +51,9 @@ class ReferralController(private val repository: ReferralRepository) {
       return ResponseEntity.badRequest().body("malformed id")
     }
 
-    return repository.findByIdOrNull(uuid)
+    return referralService.updateDraftReferral(uuid, partialUpdate)
```

Comment:
(Apologies if this is out of context)

It seems like this `update` signature does two things:
- finds a referral by ID or shortcut to null
- updates the referral and returns the updated object

To avoid having to change both `getDraftReferral` and this function when the lookup logic changes, what do you think about having one function for each of these responsibilities and composing them?